src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,421 @@
+        public string FirmwareName
+        {
+            get
+            {
+                return _firmwareName ?? string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Firmata version found on the board.
+        /// </summary>
+        public Version FirmataVersion
+        {
+            get
+            {
+                if (_protocolVersion == null)
+                {
+                    return new Version();
+                }
+
+                return _protocolVersion;
+            }
+        }
+
+        internal FirmataDevice Firmata
+        {
+            get
+            {
+                return _firmata ?? throw new ObjectDisposedException(nameof(ArduinoBoard));
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of capabilities per pin
+        /// </summary>
+        public ReadOnlyCollection<SupportedPinConfiguration> SupportedPinConfigurations
+        {
+            get
+            {
+                return _supportedPinConfigurations.AsReadOnly();
+            }
+        }
+
+        internal void Log(string message)
+        {
+            LogMessages?.Invoke(message, null);
+        }
+
+        private void FirmataOnError(string message, Exception? innerException)
+        {
+            LogMessages?.Invoke(message, innerException);
+        }
+
+        /// <summary>
+        /// Creates a GPIO Controller instance for the board. This allows working with digital input/output pins.
+        /// </summary>
+        /// <returns>An instance of GpioController, using an Arduino-Enabled driver</returns>
+        public GpioController CreateGpioController()
+        {
+            return CreateGpioController(DefaultPinNumberingScheme);
+        }
+
+        /// <summary>
+        /// Creates a GPIO Controller instance for the board. This allows working with digital input/output pins.
+        /// </summary>
+        /// <param name="pinNumberingScheme">Pin numbering scheme to use for this controller</param>
+        /// <returns>An instance of GpioController, using an Arduino-Enabled driver</returns>
+        public virtual GpioController CreateGpioController(PinNumberingScheme pinNumberingScheme)
+        {
+            return new GpioController(pinNumberingScheme, new ArduinoGpioControllerDriver(this, _supportedPinConfigurations));
+        }
+
+        /// <summary>
+        /// Creates an I2c device with the given connection settings.
+        /// </summary>
+        /// <param name="connectionSettings">I2c connection settings. Only Bus 0 is supported.</param>
+        /// <returns>An <see cref="I2cDevice"/> instance</returns>
+        /// <exception cref="NotSupportedException">The firmware reports that no pins are available for I2C. Check whether the I2C module is enabled in Firmata.
+        /// Or: An invalid Bus Id or device Id was specified</exception>
+        public virtual I2cDevice CreateI2cDevice(I2cConnectionSettings connectionSettings)
```

Comment:
should ArduinoBoard have I2cBus as property or create bus instead of devices? I know currently only bus 0 is supported but:
- assume I'd want to combine the board with SenseHat or similar device - I can't do that without implementing bus myself
- what if new Arduino shows up with 2 I2cBuses

Reply:
Some more expensive boards actually do have 2 busses (i.e. the Arduino Due), but for now the protocol doesn't support it. So this would also require a (potentially breaking) protocol extension. I don't really mind doing that (basically requiring our binding to require a specifically modified firmware), but it's out of scope here. 

About the Bus/Device destinction I think we should wait for the board PR here, as there will be some API adjustments like that anyway.